ApprovalTests/integrations/catch/Catch2Approvals.h

```diff
@@ -15,15 +17,21 @@
 #define APPROVAL_TESTS_INCLUDE_CPPS
 
 // begin-snippet: required_header_for_catch
-#include <catch2/catch.hpp>
+#include "ApprovalTests/integrations/catch/Catch2Include.h"
 // end-snippet
 
 //namespace ApprovalTests {
-struct Catch2ApprovalListener : Catch::TestEventListenerBase
+#ifdef APPROVAL_TESTS_USE_CATCH2V3
+using CatchListener = Catch::EventListenerBase;
+#else
+using CatchListener = Catch::TestEventListenerBase;
+#endif
```

Comment:
Naming has changed between catch2 v2 and catch2 v3. To support both versions this dispatch is needed.